tripal/tripal
# Fix Tripal Bulk Pub Importer Email Report

Issue #465 

## Type(s) of Change(s)


## Description


The drush command for importing publications in Tripal v2 would generate a report that got emailed to a user.  This was intended to help the site admin keep up with what pubs were being imported.   That report was not working.  This PR fixes that.   

**Do not merge this PR until #584 has been approved.  This branch actually contains the code from both PRs, so we don't want to merge this one until #584 is first merged. But please test now!!**


## Testing?





To test this PR you must have email working on your Drupal server.  

1) Go to Tripal > Data Loaders > Chado Publication Bulk Importer and create an importer if you don't already have one. I found that an easy one to test with few publications is to set the number of days to 30, and the search term to 'Citrus'. It returns just a few publications. If you have some importers already you might want to deactivate them temporarily to keep this test from taking too long.
2)  Run this drush command and be sure to set [username] to the Drupal username and [email] to the email of the person who should get the report.
```
drush trp-import-pubs --username=[username] --update=Y --report=[email]
```
When the command completes you should get an email at the address provided listing the publications that were added or updated.

## Screenshots (if appropriate):
You should receive an email similar to the following:
```
.... Tripal Bulk Publication Report

87 publication(s) were inserted:
  1) Taub DG, Awal MR, Gabel CV. O-GlcNAc Signaling Orchestrates the
     Regenerative Response to Neuronal Injury in Caenorhabditis elegans..Cell
     reports. 2018 Aug 21; 24(8):1931-1938.e3.
  2) Denzel MS, Lapierre LR, Mack HID. Emerging Topics in C. elegans aging
     research: Transcriptional regulation, stress response and epigenetics..
     Mechanisms of ageing and development. 2018 Aug 19.
  3) Hibshman JD, Leuthner TC, Shoben C, Mello DF, Sherwood DR, Meyer JN,
     Baugh LR. Non-selective autophagy reduces mitochondrial content during
     starvation in Caenorhabditis elegans.. American journal of physiology.
     Cell physiology. 2018 Aug 22.
  4) Xu F, Guang S, Feng X. Distinct nuclear and cytoplasmic machineries
     cooperatively promote the inheritance of RNAi in C. elegans.. Biology of
     the cell. 2018 Aug 22.
  5) Johnson SL, Park HY, DaSilva NA, Vattem DA, Ma H, Seeram NP.
     Levodopa-Reduced Mucuna pruriens Seed Extract Shows Neuroprotective
     Effects against Parkinson's Disease in Murine Microglia and Human
     Neuroblastoma Cells, Caenorhabditis elegans, and Drosophila
     melanogaster.. Nutrients. 2018 Aug 22; 10(9).
  6) Woodruff GC, Phillips PC. Field studies reveal a close relative of C.
     elegans thrives in the fresh figs of Ficus septica and disperses on its
     Ceratosolen pollinating wasps.. BMC ecology. 2018 Aug 21; 18(1):26.
  7) Roy D, Kahler DJ, Yun C, Hubbard EJA. Functional Interactions Between
     rsks-1/S6K, glp-1/Notch, and Regulators of Caenorhabditis elegans
     Fertility and Germline Stem Cell Maintenance.. G3 (Bethesda, Md.). 2018 
```

## Additional Notes (if any):
